toptal/picasso
# fix(Autocomplete): handle using scrollbar in options

[FX-2668]

### Description

This PR was created to fix issue in the Select component, where you cannot focus the search input by clicking on the placeholder of search input (It worked when you clicked on the border of the input).

This PR:
- reverts the [functionality](https://github.com/toptal/picasso/pull/2233) that caused the issue 
- implements a different solution to fix the issue that the [reverted PR](https://github.com/toptal/picasso/pull/2233) was trying to fix

### How to test

[TagSelector](https://picasso.toptal.net/FX-2668-fix-select/?path=/story/forms-tagselector--tagselector#default) and [Autocomplete](https://picasso.toptal.net/FX-2668-fix-select/?path=/story/forms-autocomplete--autocomplete#default)
- for both components
- go to default story and edit the options to have 14+ options (to trigger the scrollbar in options)
- when you try to scroll by dragging the scrollbar, the options will not hide

Select
- open the [story with search behavior](https://picasso.toptal.net/FX-2668-fix-select/?path=/story/forms-select--select#search-behavior)
- when clicking directly on the placeholder of the search input, it should focus the input

- clicking on the icon does not work and it is fixed in [different PR](https://github.com/toptal/picasso/pull/2691)

### Review


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-2668]: https://toptal-core.atlassian.net/browse/FX-2668?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
